lib/ui/discover/discover_child_item.dart

```diff
@@ -7,81 +7,58 @@ import 'package:farmsmart_flutter/utils/styles.dart';
 import 'package:flutter/material.dart';
 import 'package:farmsmart_flutter/utils/assets.dart';
 
+import 'discover_page.dart';
+
 class DiscoveryListItem {
-  Widget buildListItem(ArticleEntity articleData, Function getRelatedArticles) {
+  Widget buildListItem(ArticleEntity articleData, Function getRelatedArticles, HomeDiscoverPageStyle discoverStyle) {
     return GestureDetector(
-        onTap: () {
-          getRelatedArticles(articleData);
-        },
-        child: Padding(
-          key: ValueKey(articleData.title ?? Strings.noTitleString),
-          padding: Paddings.listOfArticlesPadding(),
-          child: Container(
-              height: 101,
-              child: Column(
+      onTap: () => getRelatedArticles(articleData),
```

Comment:
I don't think its this lists job to get related articles. I think the article detail widget should handle that. what if we change related articles or removed them? We may get to an article from another entry point, we would have to add the fetch at all the entry points, its better to let the article widget (the detail view) fetch repeated articles, then its self contained and can work without any dependences on other widgets fetching data it needs....

Reply:
We are doing the fetch of related articles here to handle de loading status of Article detail screen. When user taps an article, we fetch the related articles of this specific article and present the detail screen in Loading status, once we have the data the status changes to success.